### ISSUE TITLE
Could not find io.github.zxing-cpp:android:2.2.0-SNAPSHOT

### ISSUE THREAD

**Issue by ShootingKing-AM** [zxing-cpp/zxing-cpp#833]
```Execution failed for task ':sdk:extractIncludeReleaseProto'.
> Could not resolve all files for configuration ':sdk:releaseCompileClasspath'.
   > Could not find io.github.zxing-cpp:android:2.2.0-SNAPSHOT.
     Required by:
         project :sdk
```

https://s01.oss.sonatype.org/content/repositories/snapshots/io/github/zxing-cpp/android/ also has only 2.1.0 which is kinda weird

**Comment by axxel** [comment]
You are right. It is weird. In the Nexus Repository Manager I also only see the 2.1.0-SNAPSHOT version. But the latest build that should have published the 2.2.0-SNAPSHOT went through [without complaint](https://github.com/zxing-cpp/zxing-cpp/actions/runs/10772492034/job/29870246038#step:6:67).

Also @nift4 said [here](https://github.com/zxing-cpp/zxing-cpp/issues/824#issuecomment-2336480776) that it works for him. Did it at the time and now the artifact is gone? I don't know maven central / nexus / sonatype and find their UI rather confusing and user unfriendly for a layman like me.

Interestingly, the [addition](https://github.com/zxing-cpp/zxing-cpp/commit/88c66f79a01a4fe374b8624c8fc319c77308639f) of the kotlin native SNAPSHOTS has apparently worked. The relevant parts of the workflow files and the `build.gradle.kts` files are almost identical.

I'd appreciate any insight from someone who knows how this gradle -> sonatype.org stuff is supposed to work. Maybe @ISNing has some idea?

**Comment by nift4** [comment]
Was it perhaps overwritten by the kotlin native snapshots? Not too sure though

**Comment by axxel** [comment]
> Was it perhaps overwritten by the kotlin native snapshots? Not too sure though

The `artifactId` is different, so that would not make any sense to me. Can you confirm that it worked for you and now it doesn't anymore?

**Comment by ShootingKing-AM** [comment]
> > Was it perhaps overwritten by the kotlin native snapshots? Not too sure though
> 
> The `artifactId` is different, so that would not make any sense to me. Can you confirm that it worked for you and now it doesn't anymore?

Yeah it worked before just 3 days back i guess

**Comment by ISNing** [comment]
according to this metadata file: https://s01.oss.sonatype.org/content/repositories/snapshots/io/github/zxing-cpp/android/maven-metadata.xml , things seems to get clear:
1. The version `2.2.0-SNAPSHOT` was shipped 4 days ago
2. Someone who have credentials (maybe accidentally?) build and pubilshed `2.1.0-SNAPSHOT` again 2 days ago, which replaced version `2.2.0-SNAPSHOT` which is supposed to be there.

I think publish the version `2.2.0-SNAPSHOT` again can solve this.

**Comment by ISNing** [comment]
Or for some reason, at the moment of 20240912201849, the version `2.2.0-SNAPSHOT` has been removed?

The modofication time of `2.1.0-SNAPSHOT` is quite wierd, by what strategy are they set? never noticed this.

**Comment by axxel** [comment]
> things seems to get clear:

That is a very interesting observation but 'clear' is nothing to me, yet. ;). The Nexus Repository Manager tells me the 2.1.0-SNAPSHOT was uploaded and last modified in 2023. Also, no-one except me and the github repo has the credentials.

I was looking for how to remove the 2.1.0-SNAPSHOT and could not find a way. But looking for information, I saw that people have to deal with the problem of getting rid of outdated snapshots. This contradicts your theory that publishing a new one (with a different version string, even) would overwrite an older one.

I re-ran the build-android CI job and the result is that I see now 2 SNAPSHOT versions (2.1.0 and 2.2.0) in the repo manager and both show up [here](https://s01.oss.sonatype.org/content/repositories/snapshots/io/github/zxing-cpp/android/) as well as I would expect. So far so good. If only I would not have the feeling that the new version will disappear again...

**Comment by ISNing** [comment]
Yes, it's quite wired there's a version 2.1.0-SNAPSHOT published in 2023 if you haven't done that at that moment. (Had you?)

And I also noticed the time 20240912201849 is very close to the time publishing kotlin-native snapshots.

But `2.1.0-SNAPSHOT` is still a wired thing because is not related to kotlin native(it has 2.2.1-SNAPSHOT) nor android(2.2.0-SNAPSHOT)

I will give a look to process of the publish of kotlin native tomorrow

**Comment by ShootingKing-AM** [comment]
Very weird as to why it is deleted from maven

**Comment by nift4** [comment]
It's gone again?

**Comment by axxel** [comment]
> It's gone again?

Apparently. :-(. The above mentioned xml-file states a `<lastUpdated>20240919200950</lastUpdated>`  (2 days ago) for the 2.1.0-SNAPSHOT. The most recent kn snapshot publication was 2 weeks ago and the most recent android 2.2.0-SNAPSHOP publication was 4 days ago. I have no idea and hardly any motivation to "debug" this "android infrastructure" problem.

If anyone could come up with a theory, that would be highly appreciated.

**Comment by ShootingKing-AM** [comment]
After some research and https://github.com/google/dagger/issues/338.

It seems that sonatype repo will delete released versions' snapshots automatically. So i think we need to bump the version to 2.2.1, on master, (since 2.2.0 is already released on sonatype repo), then make a SNAPSHOT release for 2.2.1 using CI.

Sounds rationale as per semvar also, the master branch is not 2.2.0 anymore after many changes were pushed onto it from 2.2.0 release.

**Comment by axxel** [comment]
Thank you @ShootingKing-AM for digging that up. That makes total sense to me. In general, I don't know the next release version right after a release. Here it will be 2.3.0 but it could have been 2.2.1 or 3.0.0 as well. As alluded to [here](https://github.com/zxing-cpp/zxing-cpp/issues/824#issuecomment-2336482282), I would be happy with simply calling it `SNAPSHOT`, which might not work as @nift4 pointed out. Maybe calling it `1.0.0-SNAPSHOT` or `0.0.0-SNAPSHOT` would be good. There is no such released version and it could always contain the latest HEAD build from the CI. Comments?

**Comment by nift4** [comment]
1.0.0-SNAPSHOT or 0.0.0-SNAPSHOT would trigger false positive warnings (and also sure appear as if) that they are older than 2.1.0-SNAPSHOT. I think increasing the number to 2.2.1/2.3.0 is the way to go.

**Comment by ShootingKing-AM** [comment]
0.0.0-SNAPSHOT wouldn't work because different release versions' snapshots would collide. For eg. 2.2.1-SNAPSHOT would point to 0.0.0-SNAPSHOT and a user of the library like us would do in `if (dependencies as '0.0.0-SNAPSHOT') dependency '0.0.0-SNAPSHOT' else dependency '2.2.1'` (platonically it should be `if (dependencies as '2.2.1-SNAPSHOT') dependency '2.2.1-SNAPSHOT' else dependency '2.2.1'`) (meaning if aquiring 2.2.1-SNAPSHOT failed meaning 2.2.1 is actually released, download 2.2.1 release instead).

And going forward we would be again be having 2.3.0-SNAPSHOT would also point to 0.0.0-SNAPSHOT, would be downloaded by users (who intended to download 2.2.1's release or its snapshot). Think its better to go by sonatype semvar format by bumping version on master.

Anyways these versioning are not published changes, for you can change versioning before next release whether it is a major version change or minor version change.
